src/libmongoc/tests/test-mongoc-client-session.c

```diff
@@ -793,12 +793,10 @@ _test_end_sessions_many (bool pooled)
    /*
     * sessions were ended on the server, ten thousand at a time
     */
-   ASSERT_CMPINT (test.started_calls, ==, 2);
-   ASSERT_CMPINT (test.succeeded_calls, ==, 2);
+   ASSERT_CMPINT (test.started_calls, ==, 1);
+   ASSERT_CMPINT (test.succeeded_calls, ==, 1);
```

Comment:
Thinking more about it, I'd prefer if we kept this as a test of the behavior of `endSessions` batching 10,000 sessions at a time.

Your new test checks that we do not send `endSessions` when there are no used sessions. But we should still be testing that if the client has _used_ more than 10,000 sessions, the `endSessions` commands batch up in groups of 10,000.

I suggest adding to the loop above that creates the sessions to use the sessions in an operation. Then these assertions can remain as they were.